torchaudio/functional/functional.py

```diff
@@ -1372,6 +1372,16 @@ def resample(
 
     assert orig_freq > 0.0 and new_freq > 0.0
 
+    if not (int(orig_freq) == orig_freq and int(new_freq) == new_freq):
+        warnings.warn(
+            "Non-integer frequencies are cast to integers and may result in poor resampling quality."
+            "The algorithm is dependent on the integer ratio between `orig_freq` and `new_freq`,"
+            "so please manually convert frequencies to integer values that maintain the ratio"
+            "prior to passing them into the function"
+            "Example: To downsample a 44100 hz waveform by a factor of 8, you can use"
+            "`orig_freq=8` and `new_freq=1` instead of `orig_freq=44100` and `new_freq=5512.5`"
```

Comment:
I think we can go farther and say it will raise an error in the future.
@cpuhrsch what do you think?

Reply:
I think it's fine if we throw an exception in the next release, but I'd create a pre-emptive issue for this and link it here so that users can chime in whether they want this or not. The overall topic seems to be "non-integral sampling rates", so maybe we can link to an umbrella issue like that.

Reply:
@carolineechen 

Can you plan the future changes and make a note in #1487, then mention that in the message ?